### ISSUE TITLE
ACLs Seemingly Not Being Processed/Honored

### ISSUE THREAD

**Issue by asimons04** [authelia/authelia#3734]
# Description

Access control rules don't seem to be respected.  In this scenario, I am trying to configure a client app such that local access from LAN uses 1FA while public access from WAN requires 2FA.  i.e. If you're inside the organization network, relax 2FA requirement.

I've read through the docs, and from what I can tell, this configuration should work. However, the only policy that seems to affect an app is one explicitly set in the `identity_providers.oidc.clients.{id}.authorization_policy` configuration.  If an `authorization_policy` is left undefined for a client, it seems to default to `two_factor` for some reason (`access_control.default_policy` is set to `deny`)

I've checked and re-checked and don't seem to be able to find any error in the configuration.  Is my understanding of access control rules correct?  If so, then this may be a bug.


# Environment Info
Authelia Version:  4.36.2 from Docker image `authelia/authelia:latest`
Platform:  Docker 20.10.16 on Ubuntu 22.04
Authelia URL:  https://auth.mydomain.org
Protected App URL:  https://app.mydomain.org
Authentication Backend:  OpenLDAP
Upstream Reverse Proxy:  Nginx 1.21.3
Session Storage:  Redis
Storage:  MySQL/MariaDB

## Expected Behavior
If a user accesses the domain `app.mydomain.org` and has a remote IP address within the LAN range, as specified in `access_control.networks`, then apply the `one_factor` policy.  If the remote IP address is not within those ranges, apply the `two_factor` policy.  The OIDC client config for this application has no `authorization_policy` defined and is expected to match an ACL policy or be denied by the `default_policy`.


## Actual Behavior
When Authelia starts up, and log is set to `debug`, it shows the OIDC client registers as `two_factor` despite having no authorization policy defined in the client config.   When accessing the protected app from inside the LAN, it continues to prompt for 2FA and seems to make no distinction between internal and external requests as directed by the access control policies.

```
authelia_1  | time="2022-07-22T17:12:54Z" level=debug msg="Registering client ada35250-b198-4293-96ac-f04233c1e084 with policy two_factor (2)"
...
authelia_1  | time="2022-07-22T17:13:12Z" level=warning msg="OpenID Connect client 'ada35250-b198-4293-96ac-f04233c1e084' requires 2FA, cannot be redirected yet" method=POST path=/api/firstfactor remote_ip=192.168.1.207
```

As shown in the logs, Authelia is receiving the client's correct IP address from the upstream reverse proxy, so it seems like it has the correct remote address to use for comparison.  That IP is also within 192.168.0.0/16 which is defined in `access_control.networks.internal`.


## Relevant Configuration
Below are the relevant configuration stanzas from `configuration.yml`.  The secrets, domain, and IDs have been replaced with dummy values for this post.  The real  and dummy domains have the same structure, though.  i.e. the real app is a subdomain of the main domain.

### Nginx Proxy Headers
These headers are set on the proxy request from Nginx to Authelia.  Both `Host` and `X-Forwarded-Host` are being sent, so Authelia should be aware of the request's hostname.

```
proxy_set_header  Host                         $host;
proxy_set_header  X-Forwarded-Host   $host;
proxy_set_header  X-Real-IP                 $proxy_protocol_addr;
proxy_set_header  X-Forwarded-For     $proxy_protocol_addr;
proxy_set_header  Upgrade                   $http_upgrade;
proxy_set_header  Connection               $http_connection;
proxy_set_header  X-Forwarded-Proto   $scheme;
```

### docker-compose.yml
```
version: '3.3'
services:
  authelia:
    image: authelia/authelia
    environment:
      - AUTHELIA_IDENTITY_PROVIDERS_OIDC_ISSUER_PRIVATE_KEY_FILE=/config/jwt_keys/private.pem
    command:
      - "authelia"
      - "--config=/config/configuration.yml"
    ports:                                                                                                                                                                                   
      - 9091:9091
    volumes:
      - ./config/:/config
    depends_on:
      - redis
      - mariadb
                                                                                                                                                                                             
  mariadb:
    image: mariadb:10.7
    environment:
      - MYSQL_ROOT_PASSWORD=SuperSecretPassword!
      - MYSQL_DATABASE=Authelia
    volumes:
      - ./data/mariadb:/var/lib/mysql
      - /etc/localtime:/etc/localtime:ro


  redis:
    image: library/redis
    volumes:
      - ./data/redis:/data
```


### Sessions
Note:  I've tried both `mydomain.org` and `.mydomain.org` with the same results.  This is based on the [documentation](https://www.authelia.com/configuration/security/access-control/#domain) note "Domains in [access control rules] must be the domain configured in the [session](https://www.authelia.com/configuration/session/introduction/#domain) configuration or subdomains of that domain."
```
session:                                                                                                                                                                                     
  ## The name of the session cookie.                                                                                                                                                         
  name: mydomain_session                                                                                                                                                                   
  domain: '.mydomain.org'                                                                                                                                                                  
  same_site: lax                                                                                                                                                                             
  secret: c4ffd55f-bd6e-4f38-8e29-cb5cc7bcf358
  expiration: 1h                                                                                                                                                                             
  inactivity: 5m                                                                                                                                                                             
  remember_me_duration: 1M                                                                                                                                                                   
                                                                                                                                                                                             
  redis:                                                                                                                                                                                     
    host: redis                                                                                                                                                                              
    port: 6379                                                                                                                                                                               
    database_index: 0                                                                                                                                                                        
    maximum_active_connections: 8                                                                                                                                                            
    minimum_idle_connections: 0 
```

### Access Control
Note:  I've also tried putting the CIDR range into the rules directly instead of using the named networks; results are the same.

```
access_control:                                                                                                                                                                              
  default_policy: deny                                                                                                                                                                       
  networks:                                                                                                                                                                                  
    - name: internal                                                                                                                                                                         
      networks:                                                                                                                                                                              
        - '192.168.0.0/16'                                                                                                                                                                   
                                                                                                                                                                                             
    - name: VPN                                                                                                                                                                              
      networks: '10.10.0.0/16'                                                                                                                                                                
                                                                                                                                                                                             
  rules:                                                                                                                                                                                     
    - domain: 'app.mydomain.org                                                                                                                                                            
      policy: one_factor                                                                                                                                                                     
      networks:                                                                                                                                                                              
        - 'internal'

    - domain: 'app.mydomain.org'                                                                                                                                                            
      policy: two_factor                                                                                                                                                                     

```

### Identity Providers
The private key is provided as a secret from a file via `AUTHELIA_IDENTITY_PROVIDERS_OIDC_ISSUER_PRIVATE_KEY_FILE` environment variable.
```
identity_providers:                                                                                                                                                                          
  oidc:                                                                                                                                                                                      
    hmac_secret: LHv65HgOeyLGdDlLo5aFMcoh3NtOBQgIrvDcIGmo70Y1QjyzUYXJXLkHuvYzOqFBv9yY4cCFsSgOBfi62ZovTUkn2J5Nktd95uNo0kw4hxu6qifHnnh0eDa3pnSJhIWg
    access_token_lifespan: 1h
    authorize_code_lifespan: 1m
    id_token_lifespan: 24h
    refresh_token_lifespan: 90m
                                                                                                                                                                                             
    enable_client_debug_messages: false
    minimum_parameter_entropy: 8
    enforce_pkce: public_clients_only

    cors:                                                                                                                                                                                    
      endpoints:                                                                                                                                                                             
        - authorization                                                                                                                                                                      
        - token                                                                                                                                                                              
        - revocation                                                                                                                                                                         
        - introspection                                                                                                                                                                      
        - userinfo                                                                                                                                                                           
      allowed_origins_from_client_redirect_uris: true

    clients:                                                                                                                                                                                 
      - id: ada35250-b198-4293-96ac-f04233c1e084
        secret: 8f170ca2-63de-4992-b516-5289187b648b
        description: "My Protected App"
        public: false
        pre_configured_consent_duration: '1y'
        audience: []
        scopes:
          - openid
          - groups
          - email
          - profile
        redirect_uris:
          - https://app.mydomain.org/oauth2/callback
        grant_types:
          - refresh_token
          - authorization_code
        response_types:
          - code
        response_modes:
          - form_post
          - query
          - fragment
        userinfo_signing_algorithm: 'RS256'

```

**Comment by james-d-elliott** [comment]
> If a user accesses the domain `app.mydomain.org` and has a remote IP address within the LAN range, as specified in `access_control.networks`, then apply the `one_factor` policy. If the remote IP address is not within those ranges, apply the `two_factor` policy. The OIDC client config for this application has no `authorization_policy` defined and is expected to match an ACL policy or be denied by the `default_policy`.

Sorry this is not the intended behavior. ACL rules do not have any relationship with OIDC.

**Comment by asimons04** [comment]
Thanks for replying; that cleared up a lot of my misunderstanding.

> Sorry this is not the intended behavior. ACL rules do not have any relationship with OIDC. ACL rules apply only to forward auth scenarios.

Re-reading the docs with that in mind, everything makes _much_ more sense now.  I've since integrated Authelia with my Nginx frontend, and the ACLs and login flow are working as expected (including conditional 2FA for local/remote logins).  

I had previously had apps running through OAuth2Proxy with Authelia as the OIDC IdP (drop-in replacing Keycloak).  Now that Authelia is integrated with Nginx/configured as it was intended, I've been able to remove OA2P from my app stack and use Nginx directly with the apps now being protected by Authelia.  Very nice.  

Thanks again for your time and a great piece of software.
